### PR TITLE
WIP leaf decay

### PR DESCRIPTION
**Pull request by StianTheDark**
Wow!

### CODE REVIEW

File: TrueCraft.Core/Logic/Blocks/LeavesBlock.cs [hunk 1]
```diff
@@ -33,6 +56,24 @@ public class LeavesBlock : BlockProvider
             return new Tuple<int, int>(4, 3);
         }
 
+        public void TryDecay(Coordinates3D coords, IMultiplayerServer server, IWorld world){
```
Comment (SirCmpwn) [inline TrueCraft.Core/Logic/Blocks/LeavesBlock.cs]: Style error here, braces go on the next line

File: TrueCraft.Core/Logic/Blocks/LeavesBlock.cs [hunk 2]
```diff
@@ -33,6 +56,24 @@ public class LeavesBlock : BlockProvider
...
+                var id = world.GetBlockID(c);
+                if (id == WoodBlock.BlockID)
+                {
+                    world.SetMetadata(coords, 0x0);
+                    return;
+                }
+                else
+                {
+                    continue;
+                }
```
Comment (SirCmpwn) [inline TrueCraft.Core/Logic/Blocks/LeavesBlock.cs]: Style error here, omit braces around one-line statements.

File: TrueCraft.Core/Logic/Blocks/LeavesBlock.cs [hunk 3]
```diff
@@ -33,6 +56,24 @@ public class LeavesBlock : BlockProvider
...
         }
 
+        public void TryDecay(Coordinates3D coords, IMultiplayerServer server, IWorld world){
+            foreach (Coordinates3D a in Adjacent)
+            {
+                var c = a + coords;
+                var id = world.GetBlockID(c);
+                if (id == WoodBlock.BlockID)
+                {
+                    world.SetMetadata(coords, 0x0);
```
Comment (SirCmpwn) [inline TrueCraft.Core/Logic/Blocks/LeavesBlock.cs]: This will change the leaf type (i.e. birch, spruce, pine). Just change the one bit.

**Comment by SirCmpwn** [comment]
Let's take this chance to backport the "leaves placed by players never decay" change. Make bit 0x8 set on BlockPlaced and skip decay if that bit is set.

**Comment by StianTheDark** [comment]
Okay.

**Pull request closed without merging**